scripts/build_docs.sh

```diff
@@ -7,9 +7,57 @@ set -euo pipefail
 # Build HTML documentation for the project
 # The output will be located in docs/docs
 
+# Function to validate and determine doc-gen4 revision
+determine_doc_gen_rev() {
+    local toolchain_content
+    local first_field
+    local second_field
+    
+    # Read the lean-toolchain file
+    if [[ ! -f "lean-toolchain" ]]; then
+        echo "Warning: lean-toolchain file not found, falling back to main branch" >&2
+        echo "main"
+        return 0
+    fi
+    
+    toolchain_content=$(< lean-toolchain)
+    
+    # Extract first and second fields
+    first_field=$(echo "$toolchain_content" | cut -f1 -d:)
+    second_field=$(echo "$toolchain_content" | cut -f2 -d:)
+    
+    # Validate first field should be leanprover/lean4
```

Comment:
This comment repeats the code below it. Don't know that there's more to say here, so let's just drop it.
```suggestion
```